core/drivers/crypto/se050/adaptors/include/se050_apdu_apis.h

```diff
@@ -0,0 +1,98 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright (C) Foundries Ltd. 2020 - All Rights Reserved
+ * Author: Jorge Ramirez <jorge@foundries.io>
+ */
+
+#ifndef SE050_APDU_APIS_H_
+#define SE050_APDU_APIS_H_
+
+#include <se050.h>
+
+struct s050_scp_rotate_cmd;
+
+sss_status_t se050_factory_reset(pSe05xSession_t ctx);
+
+sss_status_t se050_cipher_update_nocache(sss_se05x_symmetric_t *ctx,
+					 const uint8_t *src, size_t src_len,
+					 uint8_t *dst, size_t *dst_len);
+
+uint8_t se050_key_exists(uint32_t k_id, pSe05xSession_t ctx);
+
+/*
+ * Key Store operations
+ */
+struct rsa_keypair_bin {
```

Comment:
Should the structs be prefixed with `se050_` like the functions above? (less risk of name clashes, consistency, self-explanatory).

Reply:
initially I had it that way but it just makes all code really messy to read....imagine if I prefix the structure, the definition below wont fit in 80 characters.....

```
sss_status_t se050_key_store_set_rsa_key_bin(sss_se05x_key_store_t *store,
					     sss_se05x_object_t *k_object,
					     struct rsa_keypair_bin *key_pair,
					     struct rsa_public_key_bin *key_pub,
					     size_t key_bit_len)
```

Reply:
How about dropping the useless `_bin` suffix? `se050_rsa_keypair` is only 2 characters longer than `rsa_keypair_bin`; on the other hand the function could be called `se050_key_store_set_rsa_key()` without `_bin` thus saving 4 characters.

Reply:
but that suffix  it is not totally useless: it is there to indicate that the format in the structure is binary (maybe it doesnt help? I thought it made sense)

Reply:
I suppose I could replace the suffix for the se050_ prefix and add a comment in the structure?